src/crawlee/static_content_crawler/_static_crawling_context.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, fields
+from typing import Generic
+
+from typing_extensions import Self, TypeVar
+
+from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction
+from crawlee._utils.docs import docs_group
+from crawlee.http_clients import HttpCrawlingResult, HttpResponse
+
+TParseResult = TypeVar('TParseResult')
+
+
+@dataclass(frozen=True)
+@docs_group('Data structures')
+class HttpCrawlingContext(BasicCrawlingContext, HttpCrawlingResult):
+    """The crawling context used by the `StaticContentCrawler`."""
+
+    @classmethod
+    def from_basic_crawling_context(cls, context: BasicCrawlingContext, http_response: HttpResponse) -> Self:
+        """Convenience constructor that creates HttpCrawlingContext from existing BasicCrawlingContext."""
+        context_kwargs = {field.name: getattr(context, field.name) for field in fields(context)}
+        return cls(http_response=http_response, **context_kwargs)
```

Comment:
Is this here just for the backward compatibility?

Reply:
This one is used quite a lot in current code base. I haven't even considered removing. It has its place as it represents context pipeline state where we already have http response, but we haven't parsed it.  So I think it is generally useful.

Reply:
Makes sense. Thanks for the explanation.